src/chording.rs

```diff
@@ -0,0 +1,288 @@
+///     { [ A B C ] }
+/// };
+///
+/// let mut layout = Layout::new(LAYERS);
+/// // Debouncer period determines chording timeout
+/// let mut debouncer: Debouncer<PressedKeys<3, 1>> =
+///     Debouncer::new(PressedKeys::default(), PressedKeys::default(), DEBOUNCE_COUNT);
+/// let mut chording = Chording::new(&CHORDS);
+///
+/// // the rest of this example should be called inside a callback
+/// // The PressedKeys are normally determined by calling the matrix
+/// // and the for loop is just to get past the debouncer
+/// for _ in 0..DEBOUNCE_COUNT {
+///     assert_eq!(0, debouncer.events(PressedKeys([[true, true, false]])).count());
+/// }
+/// let mut events = chording
+///     .tick(debouncer.events(PressedKeys([[true, true, false]])).collect())
+///     .into_iter();
+/// let event = events.next();
+/// assert_eq!(Some(Event::Press(0, 2)), event);
+/// layout.event(event.unwrap());
+/// let event = events.next();
+/// assert_eq!(None, event);
+/// ```
+use crate::layout::Event;
+use heapless::Vec;
+
+type KeyPosition = (u8, u8);
+
+/// Description of the virtual key corresponding to a given chord.
+/// keys are the coordinates of the multiple keys that make up the chord
+/// result is the outcome of the keys being pressed
+pub type ChordDef = (KeyPosition, &'static [KeyPosition]);
+
+/// Runtime data for a chord
+#[derive(Clone)]
+struct Chord {
+    def: &'static ChordDef,
+    in_progress: bool,
+    keys_pressed: Vec<bool, 8>,
+}
+
+impl Chord {
+    /// Create new chord from user data.
+    fn new(def: &'static ChordDef) -> Self {
+        let mut me = Self {
+            def,
+            in_progress: false,
+            keys_pressed: Vec::new(),
+        };
+        for _ in def.1 {
+            me.keys_pressed.push(false).unwrap()
+        }
+        me
+    }
+
+    fn tick(&mut self, events: &Vec<Event, 8>) {
+        for e in events {
+            for (k, _) in self
+                .def
+                .1
+                .iter()
+                .enumerate()
+                .filter(|(_, key)| **key == e.coord())
+            {
+                self.keys_pressed[k] = e.is_press();
+            }
+        }
+    }
+
+    fn contains_chord(&mut self, events: &Vec<Event, 8>) -> bool {
+        for key in self.def.1 {
+            if events
+                .iter()
+                .position(|&k| (&k.coord() == key && k.is_press()))
+                .is_none()
+            {
+                return false;
+            }
+        }
+        true
+    }
+
+    fn handle_chord(&mut self, events: &mut Vec<Event, 8>) {
+        self.in_progress = true;
+        for key in self.def.1 {
+            if let Some(position) = events
+                .iter()
+                .position(|&k| (&k.coord() == key && k.is_press()))
+            {
+                events.swap_remove(position);
+            }
+        }
+        events
+            .push(Event::Press(self.def.0 .0, self.def.0 .1))
```

Comment:
```suggestion
            .push(Event::Press(self.def.0.0, self.def.0.1))
```

Reply:
rustfmt wants to add the space. Wouldn't we need a rustfmt toml if we wanted something other than default?

Reply:
If rust-fmt Want  it, it know better than me ;-)